templates/kubernetes/terraform/modules/kubernetes/backend_service.tf

```diff
@@ -0,0 +1,24 @@
+data "aws_secretsmanager_secret" "by-name" {
+  name = "<% .Name %>_cf_keypair"
+}
+
+data "aws_secretsmanager_secret_version" "cf_keypair" {
+  secret_id = data.aws_secretsmanager_secret.<% .Name %>_cf_keypair.id
+}
+
+locals {
+  cf_keypair_json = jsondecode(data.aws_secretsmanager_secret_version.cf_keypair.secret_string)
+}
+
+resource "kubernetes_secret" "cf_keypair" {
+  metadata {
+    name = "cf-keypair"
+  }
+
+  data = {
+    keypair_id = local.cf_keypair_json["keypair_id"]
+    secret_key = local.cf_keypair_json["private_key"]
+  }
+
+  type = "Opaque"
+}
```

Comment:
I _think_ this should work properly this time?
I was setting an "imaginary" _value_ attribute on the **secret_version** resource previously for the decoded json.  I don't think that would have done anything meaningful.
I'm attempting to set it to a local var here - which appears to be kosher.

Reply:
Yeah, looks good! If you don't have a good testing environment I can test this stuff out for you.

Reply:
Yep, after making the changes recommended above both the script and terraform work properly.

Reply:
Thanks for working with me on this. I'm sure it must have been painful on your end! :-)

Reply:
Hehe nope, it's no problem!